.github/actions/verify-metrics-snapshot/action.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2023 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Verify Metric Snapshot and Upload Metrics'
+description: 'Upload or cache the metrics data after verification'
+inputs:
+  snapshot:
+    description: 'Path to the metric file'
+    required: true
+  artifact_key:
+    description: 'Artifact key used for uploading and fetching artifacts'
+    required: true
+  cache_key:
+    description: 'Cache key used for uploading and fetching the correct cached metric'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Upload Metrics Artifact
+      uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
+      with:
+        name: ${{ inputs.artifact_key }}
+        path: ./.metrics/${{ inputs.snapshot }}
+        retention-days: 7
+
+    - name: Cache scraped metrics for tagged release for longer retention
+      if: contains(github.ref,'tags')
+      uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57
+      with:
+        path: ./.metrics/${{ inputs.snapshot }}
+        key: ${{ github.ref_name }}_${{ inputs.artifact_key }}
```

Comment:
I was looking through our previous runs of any e2e job and I don't see any of them running on tags. It's probably because they e2e-all workflow only runs on pushes
```
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]
```

I would recommend testing some simple version of this process in a test repository.